core/src/main/java/com/sequenceiq/cloudbreak/converter/stack/cluster/gateway/GatewayTopologyJsonToGatewayTopologyConverter.java

```diff
@@ -1,12 +1,56 @@
 package com.sequenceiq.cloudbreak.converter.stack.cluster.gateway;
 
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.sequenceiq.cloudbreak.api.model.ExposedService;
 import com.sequenceiq.cloudbreak.api.model.stack.cluster.gateway.GatewayTopologyJson;
+import com.sequenceiq.cloudbreak.controller.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.controller.validation.ValidationResult;
+import com.sequenceiq.cloudbreak.controller.validation.ValidationResult.State;
+import com.sequenceiq.cloudbreak.controller.validation.Validator;
 import com.sequenceiq.cloudbreak.converter.AbstractConversionServiceAwareConverter;
+import com.sequenceiq.cloudbreak.domain.json.Json;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.gateway.ExposedServices;
 import com.sequenceiq.cloudbreak.domain.stack.cluster.gateway.GatewayTopology;
 
+@Component
 public class GatewayTopologyJsonToGatewayTopologyConverter extends AbstractConversionServiceAwareConverter<GatewayTopologyJson, GatewayTopology> {
+
+    private final Validator<GatewayTopologyJson> validator;
+
+    public GatewayTopologyJsonToGatewayTopologyConverter(Validator<GatewayTopologyJson> gatewayTopologyJsonValidator) {
+        validator = gatewayTopologyJsonValidator;
+    }
+
     @Override
     public GatewayTopology convert(GatewayTopologyJson source) {
-        return null;
+        ValidationResult validationResult = validator.validate(source);
+        if (validationResult.getState() == State.ERROR) {
+            throw new BadRequestException(validationResult.getFormattedErrors());
+        }
+        GatewayTopology gatewayTopology = new GatewayTopology();
+        gatewayTopology.setTopologyName(source.getTopologyName());
+        return convertExposedServices(gatewayTopology, source.getExposedServices());
+    }
+
+    private GatewayTopology convertExposedServices(GatewayTopology gatewayTopology, List<String> exposedServiceList) {
+        try {
+            if (!CollectionUtils.isEmpty(exposedServiceList)) {
+                ExposedServices exposedServices = new ExposedServices();
```

Comment:
This block looks (from far distance) code duplication with `GatewayJsonToGatewayConverter.convertLegacyExposedServices()`